### PR TITLE
feat(profiles): Add support for continuous profiling

### PR DESCRIPTION
**Pull request by phacops**
This will add support for profile chunks for continuous profiling.

The format used for profile chunks is the Sample Format v2, following the same processing as Sample Format v1.

For continuous profiling, we will bill using the profile duration but we're still tracking a count of chunks we processed via the `ProfileChunk` data category, mostly to count invalid chunks (that may or may not have a duration).

Related Relay PR: https://github.com/getsentry/relay/pull/3270.

### CODE REVIEW

File: src/sentry/profiles/task.py [hunk 1]
```diff
@@ -103,26 +103,37 @@ def process_profile_task(
...
-    if "event_id" not in profile:
-        profile["event_id"] = event_id
+    profile_metadata = {
+        "organization_id": profile["organization_id"],
+        "project_id": profile["project_id"],
+    }
+
+    if "profile_id" in profile:
+        profile["event_id"] = profile["profile_id"]
+        profile_metadata["profile_id"] = profile["profile_id"]
```
Comment (Zylphrex) [inline src/sentry/profiles/task.py:113]: No strong feelings here but this line is redundant since it'll be added in the next if block.

File: src/sentry/profiles/task.py [hunk 2]
```diff
@@ -144,10 +155,11 @@ def process_profile_task(
         set_measurement("profile.stacks.processed", len(profile["profile"]["stacks"]))
         set_measurement("profile.frames.processed", len(profile["profile"]["frames"]))
 
-    if options.get(
-        "profiling.generic_metrics.functions_ingestion.enabled"
-    ) and project.organization_id in options.get(
-        "profiling.generic_metrics.functions_ingestion.allowed_org_ids"
+    if (
+        profile.get("version") != "2"
```
Comment (Zylphrex) [inline src/sentry/profiles/task.py:159]: For clarity, can we put this in a helper function? Perhaps something like

```python
def get_profile_version(profile):
  version = profile.get("version")
  if version == "2":
    return 2
  # older SDKs may not set this value so assume version 1
  return 1
```

I assume it's `!= "2"` because the empty string absence of implies version 1 in some older sdks.

Same for all the calls later.

File: src/sentry/features/temporary.py [hunk 3]
```diff
@@ -186,6 +186,7 @@ def register_temporary_features(manager: FeatureManager):
     manager.add("organizations:profiling-global-suspect-functions", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
     manager.add("organizations:profiling-summary-redesign", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
     manager.add("organizations:profiling-using-transactions", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
+    manager.add("organizations:continuous-profiling", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
```
Comment (Zylphrex) [inline src/sentry/features/temporary.py:189]: you need to set a default for this feature flag in https://github.com/getsentry/sentry/blob/master/src/sentry/conf/server.py#L1452

File: src/sentry/profiles/task.py (hunk 2 above)
Comment (phacops) [reply src/sentry/profiles/task.py:159]: No version actually implies the `legacy` format.

File: src/sentry/profiles/task.py [hunk 4]
```diff
@@ -102,26 +102,37 @@ def process_profile_task(
...
+
+    if "profile_id" in profile:
+        profile["event_id"] = profile["profile_id"]
+        profile_metadata["profile_id"] = profile["profile_id"]
+
+    if "event_id" in profile:
+        profile_metadata["profile_id"] = profile["event_id"]
+
+    if "chunk_id" in profile:
+        profile_metadata["chunk_id"] = profile["chunk_id"]
```
[the remark this answers is not in the document]
Comment (phacops) [reply src/sentry/profiles/task.py:119]: That's why this line is here. `profile_metadata` is the context.

**Review by Zylphrex — APPROVED** [review]

File: src/sentry/features/temporary.py (hunk 3 above)
Comment (jjbayer) [inline src/sentry/features/temporary.py:189]: There is a chance of flakiness when you use Flagr for a feature in relay project configs: If Flagr times out, the feature defaults to "disabled". Because project configs are cached for up to one hour, this might accidentally disable the feature.

TL;DR if you need the feature to be enabled consistently, use `FeatureHandlerStrategy.INTERNAL`.

File: src/sentry/profiles/task.py [hunk 5]
```diff
@@ -321,38 +332,45 @@ def _normalize_profile(profile: Profile, organization: Organization, project: Pr
...
 
-        classification_options.update(device_options)
-        classification = str(classify_device(**classification_options))
+    classification = str(classify_device(**classification_options))
 
-        if "version" in profile:
-            profile["device"]["classification"] = classification
-        else:
-            profile["device_classification"] = classification
+    if version == "1":
```
Comment (jjbayer) [inline src/sentry/profiles/task.py:370]: In some places above, we check for `!= "2"`. Should all checks be `== "1"` to be safe & consistent?

**Review by jjbayer — APPROVED** [review]
Insert comment about test coverage here.

File: src/sentry/profiles/task.py (hunk 5 above)
Comment (phacops) [reply src/sentry/profiles/task.py:370]: Not necessarily. `!= "2"` covers Sample and Legacy formats profiles, `== "1"` would just cover the Sample format.
